src/main/java/org/zalando/nakadi/domain/storage/ZookeeperConnection.java

```diff
@@ -0,0 +1,122 @@
+package org.zalando.nakadi.domain.storage;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class ZookeeperConnection {
+
+    private final ZookeeperConnectionType type;
+
+    private final List<AddressPort> addresses;
+
+    private final String path;
+
+    public ZookeeperConnection(
+            @JsonProperty("type") final ZookeeperConnectionType type,
+            @JsonProperty("addresses") final List<AddressPort> addresses,
+            @JsonProperty("path") final String path) {
+        this.type = type;
+        this.addresses = addresses;
+        this.path = path;
+    }
+
+    public ZookeeperConnectionType getType() {
+        return type;
+    }
+
+    public List<AddressPort> getAddresses() {
+        return addresses;
+    }
+
+    @Nullable
+    public String getPath() {
+        return path;
+    }
+
+    public String getPathPrepared() {
+        if (null == path) {
+            return "";
+        } else {
+            return path.startsWith("/") ? path : ("/" + path);
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "ZookeeperConnection{" +
+                "type=" + type +
+                ", addresses=" + addresses +
+                ", path='" + path + '\'' +
+                '}';
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o)  {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        final ZookeeperConnection that = (ZookeeperConnection) o;
+        return type == that.type &&
+                Objects.equals(addresses, that.addresses) &&
+                Objects.equals(path, that.path);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(type, addresses, path);
+    }
+
+    private static final Pattern ZK_CONN_PATTERN =
+            Pattern.compile("^(\\w+)://([\\w,:\\.-]*)(/[\\w\\-\\./]*)?$");
+
+    public static ZookeeperConnection valueOf(final String connStr) {
+        final Matcher m = ZK_CONN_PATTERN.matcher(connStr);
+        if (!m.matches()) {
+            throw new RuntimeException("Unsupported format of zk connection string " + connStr);
+        }
+        final String typeString = m.group(1);
+        final Optional<ZookeeperConnectionType> optType = Arrays.stream(ZookeeperConnectionType.values())
+                .filter(v -> v.name().equalsIgnoreCase(typeString))
+                .findAny();
+        if (!optType.isPresent()) {
+            throw new RuntimeException("Can not detect type of connection from " + typeString + " in " + connStr);
+        }
```

Comment:
I think it will be easier to do 
```java
try {
    ZookeeperConnectionType.valueOf(typeString);
} catch (IllegalArgumentException e) {
    throw new RuntimeException("Can not detect type of connection from " + typeString + " in " + connStr);
}
```

Reply:
I do not want to do that because I am ignoring case here (and we have hidden configuration somewhere else, that we are using lower letters for enums).

Reply:
Ok, I see. Then I suggest to make it more elegant :)
```java
final ZookeeperConnectionType type = Arrays.stream(ZookeeperConnectionType.values())
                .filter(v -> v.name().equalsIgnoreCase(typeString))
                .findAny()
                .orElseThrow(() -> new RuntimeException("Can not detect type of connection from " + typeString + " in " + connStr));
```